plan/statistics/statistics.go

```diff
@@ -311,19 +378,191 @@ func estimateNDV(sc *variable.StatementContext, count int64, samples []types.Dat
 	return int64(estimatedDistinct), nil
 }
 
+// build4SortedColumn builds column statistics for sorted columns.
+func (t *Table) build4SortedColumn(sc *variable.StatementContext, offset int, result ast.RecordSet, bucketCount int64, isPK bool) error {
+	var id int64
+	if isPK {
```

Comment:
Yes, because PK with type integer has good properties like index columns, which is already sorted.